### ISSUE TITLE
swapped foreign keys on postgresql

### ISSUE THREAD

**Issue by sakulstra** [prisma/prisma#5924]
## Bug description
I've been building a sqlite app which worked quite fine & yesterday i thought i'll try porting it to postgresql and faced some issues with `prisma migrate dev --preview-feature`.

Apparently constraint relations got broken.

## How to reproduce

1. create these models
```sql
model Reserve {
  id                          Int               @id @default(autoincrement())
  underlyingAsset             String
  poolId                      String
  liquidations                LiquidationCall[] @relation("collateralReserve")
  liquidators                 LiquidationCall[] @relation("principalReserve")
  @@unique([underlyingAsset, poolId])
}

model LiquidationCall {
  id                  String  @id
  poolId              String
  collateralReserveId String
  collateralReserve   Reserve @relation("collateralReserve", fields: [collateralReserveId, poolId], references: [underlyingAsset, poolId])
  principalReserveId  String
  principalReserve    Reserve @relation("principalReserve", fields: [principalReserveId, poolId], references: [underlyingAsset, poolId])
}
```
2. run prisma:migrate
3. check the ouput and you'll see
`ALTER TABLE "LiquidationCall" ADD FOREIGN KEY ("poolId", "collateralReserveId") REFERENCES "Reserve"("underlyingAsset","poolId") ON DELETE CASCADE ON UPDATE CASCADE;` and similar - for some reason

## Expected behavior

The foreign keys should be in correct order as otherwise the relation breaks.
`ALTER TABLE "LiquidationCall" ADD FOREIGN KEY ("collateralReserveId", "poolId") REFERENCES "Reserve"("underlyingAsset", "poolId") ON DELETE CASCADE ON UPDATE CASCADE;`

## Prisma information
posted it above as reproduction

## Environment & setup

- OS: arch
- Database: PostgreSQL
- Node.js version: 12.14
- Prisma version: 2.17

```
prisma               : 2.17.0
@prisma/client       : 2.17.0
Current platform     : debian-openssl-1.1.x
Query Engine         : query-engine 3c463ebd78b1d21d8fdacdd27899e280cf686223 (at node_modules/@prisma/engines/query-engine-debian-openssl-1.1.x)
Migration Engine     : migration-engine-cli 3c463ebd78b1d21d8fdacdd27899e280cf686223 (at node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Introspection Engine : introspection-core 3c463ebd78b1d21d8fdacdd27899e280cf686223 (at node_modules/@prisma/engines/introspection-engine-debian-openssl-1.1.x)
Format Binary        : prisma-fmt 3c463ebd78b1d21d8fdacdd27899e280cf686223 (at node_modules/@prisma/engines/prisma-fmt-debian-openssl-1.1.x)
Studio               : 0.353.0
Done in 2.21s.
```

**Comment by tomhoule** [comment]
Hi @sakulstra , this is a bug that was fixed in 2.18.0 (released today). Thanks for reporting! Please feel free to reopen if you observe that behaviour in 2.18.0.

**Comment by tomhoule** [comment]
PR for reference: https://github.com/prisma/prisma-engines/pull/1668
